Adding l5kit and dataset installation at the top of the notebook

perone opened the pull request:
This PR will add a hidden cell (on google colab) to automatically install the sample dataset as well as l5kit. This enables the notebook to run without any problems in Google colab.

The cell looks like this:
![image](https://user-images.githubusercontent.com/412328/109875189-08bd5180-7c68-11eb-86f9-eb0cf447d2d4.png)

lucabergamini commented on examples/visualisation/visualise_data.ipynb:
```diff
@@ -19,13 +21,64 @@
+    "!wget https://lyft-l5-datasets-public.s3-us-west-2.amazonaws.com/prediction/v1.1/aerial_map.tar -q --show-progress\n",
+    "!tar xf aerial_map.tar\n",
+    "!mv aerial_map l5kit_dataset\n",
+    "!rm -rf aerial_map.tar\n",
```
don't we need also to move  `meta.json` into the root folder ? It should be inside `semantic_map.tar`

perone replied:
The semantic map is already being decompressed directly into the destination folder. Is that what you mean ?

lucabergamini replied:
isn't it decompress inside `l5kit_dataset/semantic_map` ?

perone replied:
Yes, everything from the `semantic_map.tar` will go into `l5kit_dataset/semantic_map`.

lucabergamini replied:
ah I see what's happening then!

I think we're falling back to [this](https://github.com/lyft/l5kit/blob/19b7ac8a35f50467ee8916b58e60298b5f5e7146/l5kit/l5kit/rasterization/rasterizer_builder.py#L144) or [this](https://github.com/lyft/l5kit/blob/19b7ac8a35f50467ee8916b58e60298b5f5e7146/l5kit/l5kit/rasterization/rasterizer_builder.py#L161) then

I've checked and the `meta.json` is not where it should be in fact...this could become an issue once we remove the hard-coded matrices

perone replied:
Wasn't aware of these matrices there, I think we should add another PR adding warnings upon fallback, I will open a new issue for us to keep track.

lucabergamini replied:
ideally we should remove the fallback entirely. The dataset has been updated now

lucabergamini commented on examples/visualisation/visualise_data.ipynb:
```diff
@@ -19,13 +21,64 @@
+    "\n",
+    "# Install L5Kit\n",
+    "print(\"Installing L5kit...\")\n",
+    "!pip install --progress-bar off --quiet -U l5kit pyyaml\n",
```
`pyyaml` should already been pulled from `l5kit` I think..

perone replied:
I think this is an issue w/ Colab, because it required to have this last version work there.

lucabergamini reviewed (commented):
minors, I'll give it a try later!

lucabergamini commented on examples/visualisation/visualise_data.ipynb:
```diff
@@ -19,13 +21,64 @@
+    "# Download sample zarr\n",
+    "print(\"Downloading sample zarr dataset...\")\n",
+    "!wget https://lyft-l5-datasets-public.s3-us-west-2.amazonaws.com/prediction/v1.1/sample.tar -q --show-progress\n",
+    "!tar xf sample.tar\n",
```
any chance we can get a tmp dir here?

perone replied:
Let's have a sync about that because since there are other notebooks I was thinking of moving this functionality to a Python code inside of the l5kit module.

perone commented:
@lucabergamini PTAL, I moved the script to a script file and added the temp dir. The code in the top of the notebook to setup for google colab will only run on google colab as well. To test it, you will have to change the link to download the script to [this one](https://raw.githubusercontent.com/lyft/l5kit/colab_notebook_viz/examples/setup_notebook_colab.sh), because it is pointing to the one after merge of this PR (master).

perone commented:
@lucabergamini I want to merge this soon because the viz notebook is failing on colab.

lucabergamini commented on examples/setup_notebook_colab.sh:
```diff
@@ -0,0 +1,36 @@
+wget https://lyft-l5-datasets-public.s3-us-west-2.amazonaws.com/prediction/v1.1/semantic_map.tar \
+    -q --show-progress -P $TEMP_DOWNLOAD_DIR
+mkdir -p $TEMP_DATASET_DIR/semantic_map
+tar xf $TEMP_DOWNLOAD_DIR/semantic_map.tar -C $TEMP_DATASET_DIR/semantic_map
```
can we move `meta.json` in `TEMP_DATASET_DIR`?

perone replied:
I totally forgot about it, will fix it.

perone replied:
Fixed

lucabergamini reviewed (dismissed):
minor, but LGTM

perone commented:
@lucabergamini I added the copy of the `meta.json`, it will require another review to be able to merge it.

lucabergamini reviewed (approved):
LGTM
